src/dune_rules/menhir.ml

```diff
@@ -193,20 +193,14 @@ module Run (P : PARAMS) : sig end = struct
       in
       Module.of_source ~visibility:Public ~kind:Impl source
     in
-    let modules =
-      (* The following incantation allows the mock [.ml] file to be preprocessed
-         by the user-specified [ppx] rewriters. *)
-      let mock_module =
-        Preprocessing.pp_module_as
-          (Compilation_context.preprocessing cctx)
-          name mock_module ~lint:false
-      in
-      Modules.singleton_exe mock_module
```

Comment:
Just to make sure I understand: the `singleton_exe` function contains some name mangling logic that is no longer present in the new code; this is OK because the `ocamlc -i` call only deals with the (unmangled) source file?

Reply:
Yes, I think that's the case. I've added the test to demonstrate this bug as well. To be honest, the code just didn't make sense to me when I was investigating the bug, so I refactored it. To my surprise, that was enough to fix the bug.